src/main/java/org/testng/junit/JUnitTestRunner.java

```diff
@@ -99,7 +96,13 @@ public void endTest(Test test) {
 
     org.testng.internal.TestResult tr = recordResults(test, tri);
 
-    TestListenerHelper.runTestListeners(tr, m_parentRunner.getTestListeners());
+    if (tr.getStatus() == ITestResult.STARTED) {

```

Comment:
Can we please help add a test that vets out these changes for a JUnit runner based test as well ?

Reply:
I can do it.